File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> citations) {
    int answer = 0;
    
    sort(citations.begin(),citations.end());
    
    int maxNum = citations[citations.size() - 1];
    
    for(int i = maxNum; i > 0; i--){
        if(citations.size() - (lower_bound(citations.begin(),citations.end(), i) - citations.begin()) >= i){
            answer = i;
            break;
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        int N = citations.length;
        Arrays.sort(citations);
        int max = citations[N-1];
        for(int i = 1; i < max && i <= N; i++){
            if(citations[N-i] >= i){
                answer = i;
            }
            else{
                break;
            }
        }
        
        return answer;
    }
}